packages/uniforms/src/JSONSchemaBridge.js

```diff
@@ -215,4 +219,16 @@ export default class JSONSchemaBridge extends Bridge {
   getValidator() {
     return this.validator;
   }
+
+  _distinctSchema(schema) {
+    if (schema.type === 'object') {
+      return schema;
+    }
+
+    if (schema.$ref) {
+      return {...schema, ...resolveRef(schema.$ref, schema)};
+    }
+
+    return schema;
+  }
```

Comment:
This doesn't have to be a part of the class.

Reply:
What do you mean?

Reply:
This could be extracted out of the class, same like the `resolveRef` function.